mne/channels/tests/test_standard_montage.py

```diff
@@ -73,3 +78,58 @@ def test_standard_superset():
     for key, value in m_1020._get_ch_pos().items():
         if key not in ('O10', 'O9'):
             assert_allclose(c_1005[key], value, atol=1e-4, err_msg=key)
+
+
+def _simulate_artinis_octamon():
+    """
+    Simulate artinis octamon channel data from numpy data.
+    This is to test data that is imported with missing or incorrect montage
+    info. This data can then be used to test the set_montage function.
+    """
+    data = np.random.normal(size=(16, 100))
+    ch_names = ['D1_S1 hbo', 'D1_S1 hbr', 'D1_S2 hbo', 'D1_S2 hbr',
+                'D1_S3 hbo', 'D1_S3 hbr', 'D1_S4 hbo', 'D1_S4 hbr',
+                'D2_S5 hbo', 'D2_S5 hbr', 'D2_S6 hbo', 'D2_S6 hbr',
+                'D2_S7 hbo', 'D2_S7 hbr', 'D2_S8 hbo', 'D2_S8 hbr']
+    ch_types = ['hbo', 'hbr', 'hbo', 'hbr',
+                'hbo', 'hbr', 'hbo', 'hbr',
+                'hbo', 'hbr', 'hbo', 'hbr',
+                'hbo', 'hbr', 'hbo', 'hbr']
+    sfreq = 10.  # Hz
+    info = create_info(ch_names=ch_names, ch_types=ch_types, sfreq=sfreq)
+    raw = RawArray(data, info, verbose=True)
```

Comment:
I modified the test to create from a numpy array instead of using the modified NIRx so it is less confusing.

Reply:
THIS WILL NOT WORK. You must keep the naming structure as I described. All the downstream processing and MNE functions expect this.